packages/compass-aggregations/src/modules/pipeline-builder/pipeline-ai.ts

```diff
@@ -121,6 +133,11 @@ function logFailed(errorMessage: string) {
       errorMessage,
     }
   );
+  track('AI Response Failed', () => ({
```

Comment:
If this is generic tracking for service request / response sequence, should it be happening in atlas service and not here? Same question for logging.

Reply:
I don't have a strong preference, if you'd prefer it there that's good with me. I do think we have more flexibility if we include it here instead of atlas service if we want to attach information specific to the aggregation experience, like the `editor_view_type`.

Reply:
Seems to me that if we want flexibility then it needs to be called on place and not through a generic method, and if we want consistent events for all places where the methods are used, it would make sense to move it to the service. We seem to be in some sort of in-between state right now with this code

Reply:
I'll look into a follow up refactor, merging the two `getAggregationFromUserInput` and `getQueryFromUserInput` methods and moving some more logging there.